_posts/2016-06-20-alpha-release-notes.markdown

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title:  "argparse2cwl alpha release notes"
+date:   2016-06-20 02:27:27 +0300
+comments: true
+---
+
+`argparse2cwl` forms CWL tool descriptions from programs which use `argparse` as their argument parser. 
+
+## Prerequisites ##
+`argparse2cwl` is compatible with both Python 2 & 3.
+Dependencies:
+* Jinja2 >= 2.8
+
+## Installing ##
+
+Installation process:
+
+1. Clone the source repository:
+	
+	```$ git clone https://github.com/common-workflow-language/gxargparse.git ```  
+
+2. Change directory to the root folder of the project and switch to `cwl-refactor` branch:
+
+	```$ git checkout origin/cwl-refactor```
+
+3. Run 
+
+	```
+	   $ python setup.py install
+	   $ gxargparse_check_path 
+	```
+
+## Running ##
+
+To run `argparse2cwl`, call your tool as usually without actual arguments with `--generate_cwl_tool` option:
+
+	$ python cnvkit.py batch --generate_cwl_tool
+
+Other options (which work only with `--generate_cwl_tool` provided, except for help message) are:
+
+* `-o FILENAME`, `--output_section FILENAME`: File with manually filled output section which is put to a formed CWL tool. Argparse2cwl is not very good at generating outputs, it recognizes output files only if they have type `argparse.FileType('w')`, so output section is often blank and should be filled manually.
+
+* `-go`, `--generate_outputs`: flag for generating outputs not only from arguments that are instances of `argparse.FileType('w')`, but also from every argument which contains `output` keyword in its name. For instance, argument `--output-file` with no type will also be placed to output section. However, '--output-directory' argument will also be treated like File, so generated tools must be checked carefully if when this option is selected.
+
+* `-b`, `basecommand`: command which appears in `basecommand` field in a resulting tool. It is handy to use this option when you run tool with shebang, but want `python` to be in `basecommand` field and the file amidst arguments. 
+Example:
+
+	```$ .search.py --generate_cwl_tool -b python```. 
+
+Basecommand of the formed tool will be `['python']`, and `search` will be a positional argument on position 0.
+
+* `-d`, `--directory`: directory for storing tool descriptions.
+
+* `--help_arg2cwl`: prints help message.
+
+## Argparse notes ##
+
+Some of argparse features can not be ported to CWL. 
```

Comment:
Can you refer to open issues/pullrequests on those? It will make it easier for us and the community to follow up on limitations and work to do.

@mr-c